src/components/BTable/BTable.vue

```diff
@@ -0,0 +1,320 @@
+        >
+          <i class='pagination-icon ion-arrow-right-c' aria-hidden='true'></i>
+        </button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+
+export default {
+
+  name: 'b-table',
+  props: {
+    /**
+    * The column headers for the table.
+    */
+    headers: {
+      type: Array,
+      required: true
+    },
+    /**
+    * The table data that will be displayed, and array of strings and/or objects.
+    */
+    tableData: {
+      type: Array,
+      required: true
+    },
+    /**
+    * Preload a page if provided.  Defualts to zero if not provided.
+    */
+    pageNumber: {
+      type: Number,
+      required: false
+    },
+    /**
+    * The number of rows to be displayed in the table per page. Defualts to 10 if not provided.
+    */
+    rowsPerPage: {
+      type: Number
+    },
+    /**
+    * The total number of rows for the table.
+    */
+    totalResults: {
+      type: Number,
+      required: true
+    },
+    /**
+    * The different row sizes availablt to the user, will default to [10, 20, 50] if not provided.
+    */
+    rowDisplayOptions: {
+      type: Array,
+    }
+  },
+  data () {
+    return {
+      rows: this.buildRows(this.tableData, this.headers),
+      unsorted: require('../../../static/unSortedArrows.svg'),
+      currentPage: this.pageNumber ? this.pageNumber : 0,
+      options: this.rowDisplayOptions ? this.rowDisplayOptions : [10, 20, 50],
+      currentRowsPerPage: this.rowsPerPage ? this.rowsPerPage : 10,
+      sorted: {},
+      id: null
+    }
+  },
+  mounted() {
+    this.id = this._uid
+  },
+  computed: {
+    /**
+    * @returns{String}
+    * This builds the string used in the table to show meta information on the table rows.
+    */
+    results: function () {
+      const start = (this.currentPage * this.currentRowsPerPage) + 1
+      let end = (start + parseInt(this.currentRowsPerPage)) - 1
+      end = end > this.totalResults ? this.totalResults : end
+      return `${start}-${end} of ${this.totalResults}`
+    }
+  },
+  methods: {
+    /**
+    * @param {Array<Object|String>} rowData
+    * @param {Array<String>} headers
+    * @returns {Array<String>}
+    * This maps the rowData with the applicable headers to form the rows.
+    * If the render key is present, a supplied render funtion will be called
+    * to populate that cell instead of a string.
+    */
+    buildRows: function(rowData, headers) {
+      let rows = []
+      rowData.forEach( function(data) {
+        let row = []
+        headers.forEach( function(header) {
+          if (header.render) {
+            row.push(header.render(data[header.key]))
+          } else {
+            row.push(data[header.key])
+          }
+        })
+        rows.push(row)
+      })
+      return rows
+    },
+    /**
+    * @param {String} sortBy
+    * Populates an object to track sort state and emits
+    * that object to the 'sort' event which a parent
+    * component can listen to using '.$on'
```

Comment:
- Can't get it to sort
![sortme](https://user-images.githubusercontent.com/18075411/36623862-3ecd402c-18c6-11e8-8026-fc6a04fab593.gif)

- also the whole header/table shifts on click

Reply:
what are you testing this on?  It should't sort, sorting will be handled by the user.